test/demo/utils/notificationsRemoteExecution.ts

```diff
@@ -0,0 +1,57 @@
+import {Identity} from 'openfin/_v2/main';
+
+import {NotificationOptions, NotificationEvent} from '../../../src/client';
+
+import {OFPuppeteerBrowser, BaseWindowContext} from './ofPuppeteer';
+
+export interface NotifsTestContext extends BaseWindowContext {
+    notifications: typeof import('../../../src/client')
+}
+
+const ofBrowser = new OFPuppeteerBrowser<NotifsTestContext>();
+
+export async function create(executionTarget: Identity, options: NotificationOptions) {
+    return ofBrowser.executeOnWindow(executionTarget, function(optionsRemote: NotificationOptions){
```

Comment:
These functions don't declare that `this` is a different context. Does TypeScript pick that up from the `executeOnWindow` declaration?..

Reply:
It picks it up from the generic type of the ofBrowser. `executeOnWindow` can be passed a separate context, but uses the one passed with the constructor by default.

Reply:
More specifically, it can be passed any context which extends the generic one. The one passed to ofBrowser is intended as the base context